frontend/src/components/eMIB/EditActionDialog.jsx

```diff
@@ -36,20 +30,36 @@ const styles = {
   }
 };
 
-class EditEmailActionDialog extends Component {
+class EditActionDialog extends Component {
   static propTypes = {
+    emailId: PropTypes.number.isRequired,
     showDialog: PropTypes.bool.isRequired,
     handleClose: PropTypes.func.isRequired,
-    saveEmail: PropTypes.func.isRequired,
-    actionType: PropTypes.oneOf([ACTION_TYPE.email, ACTION_TYPE.task]).isRequired,
-    editMode: PropTypes.oneOf([EDIT_MODE.create, EDIT_MODE.update]).isRequired
+    actionType: PropTypes.oneOf(Object.values(ACTION_TYPE)).isRequired,
+    editMode: PropTypes.oneOf(Object.values(EDIT_MODE)).isRequired,
+    // Provided from Redux.
+    addEmail: PropTypes.func.isRequired,
+    addTask: PropTypes.func.isRequired
+  };
+
+  state = {
+    action: {}
   };
 
   handleSave = () => {
-    this.props.saveEmail();
+    if (this.props.actionType === ACTION_TYPE.email) {
+      this.props.addEmail(this.props.emailId, this.state.action);
+    } else if (this.props.actionType === ACTION_TYPE.task) {
+      this.props.addTask(this.props.emailId, this.state.action);
+    }
     this.props.handleClose();
   };
 
+  // updatedAction is of shape actionShape
```

Comment:
I find this comment a little confusing. Can you add more details on this one?

Reply:
Fixed in commit #https://github.com/code-for-canada/project-thundercat/pull/180/commits/3c9c21dcb1d501835c17bfb6dc4e00a4fb95dc12